docs/cql/tombstone-gc.rst

```diff
@@ -0,0 +1,109 @@
+
+* Repair-based Tombstone GC (recommended) - tombstones are removed when a repair 
+  process has been performed.
+* Timeout-based Tombstone GC - tombstones are removed when they exceed the time 
+  specified with the ``gc_grace_seconds`` option.
+
+
+Repair-based Tombstone GC (recommended)
+-----------------------------------------
+
+When repair-based tombstone GC is enabled, tombstones are only removed after repair 
+is performed. Repair synchronizes data between nodes so that all replicas hold 
+the same data. In this way, repair-based tombstone GC prevents data resurrection.
+
+You can enable repair-based tombstone GC when you CREATE TABLE or ALTER TABLE. 
+Set the ``tombstone_gc`` option to ``repair``. For example:
+
+.. code:: cql
+
+    CREATE TABLE ks.cf (key blob PRIMARY KEY,  val blob) WITH tombstone_gc = {'mode':'repair'};
+
+.. code:: cql
+
+    ALTER TABLE ks.cf WITH tombstone_gc = {'mode':'repair'} ;
+
+See :ref:`Tombstones GC options <ddl-tombstones-gc>`.
+
+Timeout-based Tombstone GC
+----------------------------
+
+When timeout-based tombstone GC is enabled, tombstones are only removed after 
+they exceed the time specified with the ``gc_grace_seconds`` option. 
+However, you must run the repair process to ensure that replicas hold the same 
+data, and the process must be completed within the ``gc_grace_seconds`` limit. 
+Failing to do so may result in the resurrection of deleted data.
+
+Compared to repair-based tombstone GC, timeout-based tombstone GC is a less 
+robust solution because:
+
+* It relies on the user starting repair within ``gc_grace_seconds``.
+* It relies on the system completing repair within ``gc_grace_seconds``. If there 
+  are other tasks in progress, repair may be slowed down and may not be completed 
+  within ``gc_grace_seconds``.
+
+To apply timeout-based tombstone GC:
+
+#. Set the ``tombstone_gc`` option to ``timeout`` when you CREATE TABLE or ALTER TABLE.
+   (You can skip this step if the timeout mode is the default; see :ref:`Tombstones GC options <ddl-tombstones-gc>`.)
+
+#. Specify the value of ``gc_grace_seconds`` when you CREATE TABLE or ALTER TABLE. The default is 864000 seconds (10 days). Example:
+
+  .. code:: cql
+
+    ALTER TABLE ks.cf WITH gc_grace_seconds = 80000; 
+
+See :doc:`How to Change gc_grace_seconds for a Table </kb/gc-grace-seconds>` for additional information.
+
+Other Tombstone GC Modes
+---------------------------
+In rare cases, you may want to choose one of the following tombstone GC modes:
+
+* ``disable`` - Tombstone GC is never performed. This mode may be useful when 
+  loading data to the database to avoid tombstone GC when some data is not 
+  yet available.
+* ``immediate`` - Tombstone GC is immediately performed. There is no wait time 
+  or repair requirement. This mode is useful for a table that uses the TWCS 
+  compaction strategy with no user deletes. After data expires after TTL, 
+  ScyllaDB can perform compaction to drop the expired data immediately.
```

Comment:
TWCS is just as vulnerable to resurrection as other compaction strategies, so I assume that there's an hidden assumption here that a repair does happen before the compaction? If that's the case, how is "immediate" better than "repair" mode?
@asias maybe you can clarify?

Reply:
The idea is that if the table uses TWCS and there is no user deletes. Users can use tombstone_gc =  immediate mode. There is no repair needed before tombstone gc at all. Today, people use  a hack with gc_grace_seconds zero for that use case.